### ISSUE TITLE
This function has been deprecated in favor of IonicCordova.getAppDetails.

### ISSUE THREAD

**Issue by uptownjimmy** [ionic-team/cordova-plugin-ionic#209]
> This function has been deprecated in favor of IonicCordova.getAppDetails.
> IonicCordova.getAppInfo @ common.js:1287

This is only happening when we deploy to a device and then monitor the app in the “chrome://inspect/#devices” console. This warning does not show when running ionic serve and debugging in Chrome.

The code printing this is in plugins/cordova-plugin-ionic/dist/common.js.

`var IonicCordova = /** @class */ (function () {
    function IonicCordova() {
        this.deploy = new IonicDeploy(this);
    }
    IonicCordova.prototype.getAppInfo = function (success, failure) {
        console.warn('This function has been deprecated in favor of IonicCordova.getAppDetails.');
        this.getAppDetails().then(function (result) { return success(result); }, function (err) {
            typeof err === 'string' ? failure(err) : failure(err.message);
        });
    };
    IonicCordova.prototype.getAppDetails = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, new Promise(function (resolve, reject) {
                        cordova.exec(resolve, reject, 'IonicCordovaCommon', 'getAppInfo');
                    })];
            });
        });
    };
    return IonicCordova;
}());`

**Comment by jcesarmobile** [comment]
Are you calling `getAppInfo()` somewhere in your code?

**Comment by uptownjimmy** [comment]
@jcesarmobile No, we are not. My assumption was that it's being called from within cordova-plugin-ionic or from another package, but I haven't had time to investigate down that road.

**Comment by justjoeyuk** [comment]
@jcesarmobile This is absolutely unacceptable. I have this issue, and I've had nothing but issues with this plugin. I pay for this service, I pay a fairly large sum for what it offers. However, I've not even managed to get it working and I'm hardly incompetent when it comes to software development.

If you checked your plugin, you'd know it is calling getAppInfo and then falling apart because you guys deprecated it. This is a "professional" tool that you expect 100's of thousands of people to rely on for content updates - please treat it like one and start actively responding to support requests because if this isn't sorted soon, I'll be straight up cancelling the subscription as it's not even nearly deserved.

**Comment by jcesarmobile** [comment]
@justjoeyuk I've tried to reproduce several times without success, and I've also checked the code multiple times and I can't find a single call to `getAppInfo()` in this plugin's code. Since looks like you have found it, can you point me to it? Anyway, `getAppInfo` internally just shows the warning and then call `getAppDetails`, so shouldn't prevent the app from working, just show the warning so people can know about the deprecation and migrate before it's removed in a future release.

About support not answering your support request, we can't find any support request from you, can you create a new support request and reference this issue and mention to forward the information to me?

**Comment by StevenH86** [comment]
I have issues now where my current build just takes it upon itself to run the deploy function and download the latest files on it's own... Previously it would only run depending on being triggered from our database. The update method is set to none but it ignores that.

I updated to the newest version of pro and now I can't build the app because of this error:

typescript: node_modules/@ionic/pro/dist/src/services/deploy/index.d.ts, line: 120 
            Cannot find name 'CheckDeviceResponse'. 

     L119:  checkForUpdate(): Promise<CheckDeviceResponse>;
     L121:   * @description Remove the files specific to a snapshot from the device.

**Comment by timbru31** [comment]
This could be caused by the https://www.npmjs.com/package/@ionic/pro package, too. It has some calls to `window.IonicCordova.getAppInfo`. Unfortunately it's still not open source so we can't provide a fix for it but still require it for the Ionic Pro Error Monitoring functionality.
